src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -486,6 +585,373 @@ public List<UserType> fetchUserTypes() {
     public List<RoleType> getJobTitles() {
         return roleTypeRepository.findAll();
     }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public StaffProfileCreationResponse updateStaffProfile(StaffProfileCreationRequest profileRequest) {
+
+        log.info("{}:: processStaffProfileCreation starts::",
+                loggingComponentName);
+
+        jsrValidatorStaffProfile.validateStaffProfile(profileRequest,STAFF_PROFILE_UPDATE);
+
+        checkStaffProfileForUpdate(profileRequest);
+
+        List<StaffProfileCreationRequest> cwUiRequests = Collections.singletonList(profileRequest);
+
+
+
+        List<CaseWorkerProfile> caseWorkerProfiles =  updateStaffProfiles(cwUiRequests);
+        StaffProfileCreationResponse response = new StaffProfileCreationResponse();
+        CaseWorkerProfile caseWorkerProfile = caseWorkerProfiles.get(0);
+
+
+        if (null != caseWorkerProfile) {
+
+            staffProfileAuditService.saveStaffAudit(AuditStatus.SUCCESS, StringUtils.EMPTY,
+                    caseWorkerProfile.getCaseWorkerId(),profileRequest,STAFF_PROFILE_UPDATE);
+            response = StaffProfileCreationResponse.builder()
+                    .caseWorkerId(caseWorkerProfile.getCaseWorkerId())
+                    .build();
+        }
+        return response;
+    }
+
+    private void checkStaffProfileForUpdate(StaffProfileCreationRequest profileRequest) {
+
+        // get all existing profile from db (used IN clause)
+        CaseWorkerProfile caseWorkerProfile = caseWorkerProfileRepo.findByEmailId(profileRequest.getEmailId());
+        if (caseWorkerProfile == null) {
+            staffProfileAuditService.saveStaffAudit(AuditStatus.FAILURE,PROFILE_NOT_PRESENT_IN_DB,
+                    StringUtils.EMPTY,profileRequest,STAFF_PROFILE_UPDATE);
+            throw new ResourceNotFoundException(PROFILE_NOT_PRESENT_IN_DB);
+        }
+    }
+
+
+    public List<CaseWorkerProfile> updateStaffProfiles(List<StaffProfileCreationRequest> cwUiRequests) {
+
+        List<CaseWorkerProfile> processedCwProfiles;
+        Map<String, StaffProfileCreationRequest> emailToRequestMap = new HashMap<>();
+
+        //create map for input request to email
+        for (StaffProfileCreationRequest request : cwUiRequests) {
+            emailToRequestMap.put(request.getEmailId().toLowerCase(), request);
+        }
+
+        try {
+            // get all existing profiles from db (used IN clause)
+            List<CaseWorkerProfile> cwDbProfiles = caseWorkerProfileRepo.findByEmailIdIn(emailToRequestMap.keySet());
```

Comment:
You need not again call the database to get the profile detail. Can you please use the response from https://github.com/hmcts/rd-caseworker-ref-api/pull/601/files#diff-1e81f87fca954d43c994f1f944b28ade4e446be6a0c98654292374200f92066aR623

Reply:
done